### PR TITLE
analyze_setup(...) small fix and feature

### PR DESCRIPTION
**Pull request by nikolarobottesla**
Included bug fix from previous PR
1. Fixed Bug in anaylyze_setup
2. Added NumVariationsToDistribute to analyze_setup

### CODE REVIEW

File: pyaedt/application/Analysis.py [hunk 1]
```diff
@@ -1642,6 +1643,8 @@ def analyze_setup(self, name, num_cores=None, num_tasks=None, num_gpu=None, acf_
             Full path to custom ACF file. The default is ``None.``
         use_auto_settings : bool, optional
             Either if use or not auto settings in task/cores. It is not supported by all Setup.
+        num_variations_to_distribute: int, optional
```
Comment (maxcapodi78) [inline pyaedt/application/Analysis.py:1646]:
```suggestion
        num_variations_to_distribute : int, optional
```

File: pyaedt/application/Analysis.py [hunk 2]
```diff
@@ -1625,7 +1625,8 @@ def get_object_material_properties(self, object_list=None, prop_names=None):
         return dict
 
     @pyaedt_function_handler()
-    def analyze_setup(self, name, num_cores=None, num_tasks=None, num_gpu=None, acf_file=None, use_auto_settings=True):
+    def analyze_setup(self, name, num_cores=None, num_tasks=None, num_gpu=None, acf_file=None, use_auto_settings=True,
+                      num_variations_to_distribute=None):
```
Comment (maxcapodi78) [inline pyaedt/application/Analysis.py:1629]:
```suggestion
    def analyze_setup(self,
                      name,
                      num_cores=None,
                      num_tasks=None,
                      num_gpu=None,
                      acf_file=None,
                      use_auto_settings=True,
                      num_variations_to_distribute=None):
```

**Review by maxcapodi78 — APPROVED** [review]
LGTM. Thanks

File: pyaedt/application/Analysis.py [hunk 3]
```diff
@@ -1642,6 +1643,8 @@ def analyze_setup(self, name, num_cores=None, num_tasks=None, num_gpu=None, acf_
             Full path to custom ACF file. The default is ``None.``
         use_auto_settings : bool, optional
             Either if use or not auto settings in task/cores. It is not supported by all Setup.
+        num_variations_to_distribute: int, optional
+            Number of variations to distribute. For this to take effect use_auto_settings must be True.
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:1647]:
```suggestion
            Number of variations to distribute. For this to take effect ``use_auto_settings`` must be set to ``True``.
```

File: pyaedt/application/Analysis.py [hunk 4]
```diff
@@ -1695,6 +1704,8 @@ def analyze_setup(self, name, num_cores=None, num_tasks=None, num_gpu=None, acf_
             if self.design_type == "Icepak":
                 use_auto_settings = False
             update_hpc_option(target_name, "UseAutoSettings", use_auto_settings, False)
+            if num_variations_to_distribute:
+                update_hpc_option(target_name, 'NumVariationsToDistribute', num_variations_to_distribute, False)
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:1708]: @nikolarobottesla This should fix the failing action named "style".
```suggestion
                update_hpc_option(target_name, "NumVariationsToDistribute", num_variations_to_distribute, False)
```

Comment (MaxJPRey) [reply pyaedt/application/Analysis.py:1708]: @nikolarobottesla Thanks again for contributing to the project.

**Review by MaxJPRey — APPROVED** [review]
LGTM.

**Comment by nikolarobottesla** [comment]
@MaxJPRey Should I move the tabs as shown in the style check?  Do I need to add a test?

File: pyaedt/application/Analysis.py [hunk 5]
```diff
@@ -1625,7 +1625,14 @@ def get_object_material_properties(self, object_list=None, prop_names=None):
...
     @pyaedt_function_handler()
-    def analyze_setup(self, name, num_cores=None, num_tasks=None, num_gpu=None, acf_file=None, use_auto_settings=True):
+    def analyze_setup(self,
+                      name,
+                      num_cores=None,
+                      num_tasks=None,
+                      num_gpu=None,
+                      acf_file=None,
+                      use_auto_settings=True,
+                      num_variations_to_distribute=None):
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:1635]:
```suggestion
    def analyze_setup(
        self,
        name,
        num_cores=None,
        num_tasks=None,
        num_gpu=None,
        acf_file=None,
        use_auto_settings=True,
        num_variations_to_distribute=None,
    ):
```

**Comment by MaxJPRey** [comment]
> @MaxJPRey Should I move the tabs as shown in the style check? Do I need to add a test?

@nikolarobottesla No, everything is fine. Thanks again.

**Pull request merged**